packages/react-flex/Spec.md

```diff
@@ -0,0 +1,524 @@
+| `grow`   | boolean \| number \| string = "`inherit`", "`initial`", "`unset`"                               | `0`           | Abstraction of `flex-grow` and override of Flex's `grow`. A `true` value will translate into `1` for ease of use and retro-compatibility with Stack.     |
+| `shrink` | boolean \| number \| string = "`inherit`", "`initial`", "`unset`"                               | `1`           | Abstraction of `flex-shrink` and override of Flex's `shrink`. A `true` value will translate into `1` for ease of use and retro-compatibility with Stack. |
+| `push`   | boolean                                                                                         | `false`       | Defines an auto margin depending on the flex direction. Kept for retro-compatibility but perhaps can be removed.                                         |
+| `basis`  | string (_see [flex-basis](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-basis#values)_) | "`auto`"      | For clarity, renaming of v0 Flex's `size` prop to `basis`.                                                                                               |
+| `order`  | number                                                                                          | `0`           | Abstraction of `order`.                                                                                                                                  |
+
+#### _Deprecating_
+
+| Original Component | Name            | Comments                                               |
+| ------------------ | --------------- | ------------------------------------------------------ |
+| StackItem          | `disableShrink` | Extraneous prop given the existance of `shrink`.       |
+| StackItem          | `verticalFill`  | Redundant. Same as a style override of `height: 100%`. |
+| FlexItem           | `size`          | Renamed to `basis`.                                    |
+
+## Sample Code
+
+### Basic
+
+```HTML
+<Flex>
+ <span>1</span>
+ <span>2</span>
+</Flex>
+```
+
+### Direction
+
+```HTML
+<Flex direction="column-reverse">
+  <span>bottom</span>
+  <span>top</span>
+</Flex>
+```
+
+### Alignment
+
+```HTML
+<Flex verticalAlign="center">
+  <span>vertically centered</span>
+  <span>vertically centered</span>
+</Flex>
+```
+
+```HTML
+<Flex horizontalAlign="center">
+  <span>horizontally centered</span>
+  <span>horizontally centered</span>
+</Flex>
+```
+
+### Spacing
+
+```HTML
+<Flex gap="5rem">
+  <span>5rem around me</span>
+  <span>5rem around me</span>
+</Flex>
+```
+
+### Wrapping
+
+```HTML
+<Flex wrap>
+  <span>1</span>
+  <span>2</span>
+</Flex>
+```
+
+### Order
+
+```HTML
+<Flex>
+  <span>2</span>
+  <span order={1}>1</span>
```

Comment:
**q:** Do I understand this correctly?  

From implementation perspective, Flex will traverse all its children to look for "special words" and apply styles on those nodes accordingly ?

or it's a typo and there should be `<FlexItem order={1}>...</>` ?

> Note: this questions applies to all examples with similar pattern

Reply:
Unsure if it's a typo or my subconscious' favoritism leaning towards a simpler usage. 😄 I can't think of a non-cumbersome way of implementing this. What do you think, would it make sense to explore this approach or keep `Flex.Item`?

Reply:
Similar thoughts here, I was looking for a while for `grow`, `order` and `shrink` DOM attributes. I also thought it was a type for `<FlexItem />`

Reply:
If it's not intuitive then I will revert to the original situation or using `<FlexItem>`. Thanks for the feedback